### PR TITLE
feat(widgets): Initial StateMachineWidget implementation

### PR DESCRIPTION
**Pull request by Sofia-Rommel**
Initial implementation of the State machine widget

### CODE REVIEW

File: src/widgets/state-widget/widget.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+				<Heading level={2}>State Machine State </Heading>
+
+				<Divider size="M" marginY="single-line-height" />
+
+				<Text marginY="size-50"> Current State: </Text>
+
+				<Flex gap="size-400">
+					<Content marginY="size-100" UNSAFE_style={{ fontSize: '1.5em' }}>
+						{' '}
+						{state}{' '}
```
Comment (pklaschka) [inline src/widgets/state-widget/widget.tsx:29]: I believe this is a change auto-introduced by Prettier when formatting the file 😉 

```suggestion
						{state}
```

File: src/widgets/state-widget/widget.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+					</Content>
+					<Badge special={true}> {Tagname} </Badge>
+				</Flex>
+
+				<Divider size="M" marginY="single-line-height" />
+
+				<Text marginY="size-50"> Next State: </Text>
+				<Content marginY="size-100" UNSAFE_style={{ fontSize: '1.5em' }}>
+					{' '}
+					{nextState}{' '}
```
Comment (pklaschka) [inline src/widgets/state-widget/widget.tsx:39]: Same as above: Probably introduced when formatting the file using Prettier

```suggestion
					{nextState}
```

**Review by pklaschka — CHANGES_REQUESTED** [review]
First of all: This looks really good. Of course, there are some minor details that'll come with experience, but for your first widget: Superb work so far 🙂. Two minor issues I've found on my first review pass (see my inline comments). Maybe, you can fix those when working on the widget the next time 🙂

File: src/widgets/state-widget/Badge.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	backgroundColor: ViewProps['backgroundColor'];
+}
+
+export function Badge({
+	special,
+	children
+}: {
+	special: boolean;
+	children: ReactNode;
+}): JSX.Element {
```
Comment (fussel178) [inline src/widgets/state-widget/Badge.tsx:17]:
```suggestion
}} {
```
Please remove the explicit return type of the component. :wink: 
See https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components for more information.

Comment (Sofia-Rommel) [reply src/widgets/state-widget/Badge.tsx]: I tried applying your suggested change and I got an error (Unexpected token, expected "," (17:1)). Then I also get this error: 'Badge', which lacks return-type annotation, implicitly has an 'any' return type.ts(7010) and I am not sure what I have to do now. What can I do to fix this problem or is there anything I can do to get it working with your suggested change?

Comment (pklaschka) [reply src/widgets/state-widget/Badge.tsx]: @Sofia-Rommel Could you quickly jump into Discord?

**Review by pklaschka — APPROVED** [review]

**Comment by pklaschka** [comment]
We can introduce the correct channel post-merge, so on my end, this is ready to merge. @fussel178 What do you think?

**Comment by pklaschka** [comment]
Manually merged with commit ee2e980cb09b2f82cf604ddf052b722e8ddabd00

To add further updates to this, please create a new branch, based on `main` 🙂 .

Thank you very much for your contribution, great work 🎉

**Pull request closed without merging**